src/commands/apextests/list.ts

```diff
@@ -20,15 +23,96 @@ export default class ApextestsList extends SfCommand<ApextestsListResult> {
       char: 'd',
       required: false,
     }),
+    format: Flags.string({
+      summary: messages.getMessage('flags.format.summary'),
+      description: messages.getMessage('flags.format.description'),
+      char: 'f',
+      required: false,
+    }),
+    // TODO: add manifest flag
   };
 
+  private static parseTestsNames(data: string): string[] {
+    // remove the prefix @Tests or @TestSuites
+    return data
+      .split(',')
+      .map((line) => line.replace(/(@Tests|@TestSuites)/, ''))
+      .map((line) => line.replace(':', ''))
+      .map((line) => line.trim())
+      .filter((line) => line.trim().length > 0);
+  }
+
+  private static listTestsInDirectory(directory: string): string[] {
+    // check if the provided directory exists
+    if (!directory) {
+      throw new Error('Invalid directory.');
+    }
+
+    let readDir;
+
+    try {
+      readDir = fs.readdirSync(directory, { recursive: true }) as string[];
+    } catch (error) {
+      throw new Error('Invalid directory.');
+    }
+
+    const files = readDir.filter((file) => file.endsWith('.cls') || file.endsWith('.trigger'));
+    const testMethodsNames: string[] = [];
+
+    // read each file and check for the test methods at the top
+    files.map((file) => {
+      const data = fs.readFileSync(`${directory}/${file}`, 'utf8');
+
+      // try to find, with a RegEx, the test methods listed at the top of the
+      // file with @Tests or @TestSuites
+      const testMethods = data.match(/(@(Tests|TestSuites)).+/g);
+      // for each entry, parse the names
+      // const testMethodsNames = testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : [];
+      testMethodsNames.push(...(testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : []));
+    });
+
+    return testMethodsNames;
+  }
+
+  private static async formatList(format: string, tests: string[]): Promise<ApextestsListResult> {
```

Comment:
**remarks:** it should not be async right ?

There is no await here and we do not need to have asynchronism ?

Reply:
You are right. I'm going to fix this and the others in a separate PR.